assets/src/edit-story/elements/video/frame.js

```diff
@@ -14,14 +14,127 @@
  */
 
+/**
+ * External dependencies
+ */
+import { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { CSSTransition } from 'react-transition-group';
+
 /**
  * Internal dependencies
  */
 import StoryPropTypes from '../../types';
 import MediaFrame from '../media/frame';
+import { elementFillContent } from '../shared';
+import { useStory } from '../../app/story';
+import { ReactComponent as PlayIcon } from '../../icons/play.svg';
+import { ReactComponent as PauseIcon } from '../../icons/pause.svg';
+
+const Play = styled(PlayIcon)`
+  width: 48px;
+  height: 48px;
+`;
+const Pause = styled(PauseIcon)`
+  width: 48px;
+  height: 48px;
+`;
+
+const Wrapper = styled.div`
+  ${elementFillContent}
+`;
+
+const ButtonWrapper = styled.div`
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+  cursor: pointer;
+
+  opacity: 0;
+  &.button-enter {
+    opacity: 0;
+  }
+  &.button-enter-active,
+  &.button-enter-done {
+    opacity: 1;
+    transition: opacity 200ms;
+  }
+  &.button-exit {
+    opacity: 1;
+  }
+  &.button-exit-active,
+  &.button-exit-done {
+    opacity: 0;
+    transition: opacity 200ms;
+  }
+`;
 
 function VideoFrame({ element }) {
-  return <MediaFrame element={element} />;
+  const { id } = element;
+  const [hovering, setHovering] = useState(false);
+  const videoNode = document.getElementById(`video-${id}`);
+  const [isPlaying, setIsPlaying] = useState(false);
+  const {
+    state: { selectedElementIds },
+  } = useStory();
+  const isElementSelected = selectedElementIds.includes(id);
+  const noElementsSelected = !selectedElementIds.length;
+
+  const onPointerEnter = () => {
+    // Sync UI for auto-play on insert.
+    const currentlyPlaying = videoNode && !videoNode.paused;
+    if (currentlyPlaying && !isPlaying) {
+      setIsPlaying(true);
+    }
+    setHovering(true);
+  };
+
+  useEffect(() => {
+    if ((!isElementSelected || noElementsSelected) && videoNode) {
+      videoNode.pause();
+      videoNode.currentTime = 0;
+      setIsPlaying(false);
+    }
+  }, [isElementSelected, noElementsSelected, videoNode]);
+
+  const handlePlayPause = () => {
+    if (isPlaying) {
+      videoNode.pause();
+      setIsPlaying(false);
+    } else {
+      videoNode.play().then(() => {
+        setIsPlaying(true);
+      });
+    }
+  };
+
+  useEffect(() => {
+    if (!videoNode) {
+      return undefined;
+    }
+
+    const onVideoEnd = () => {
+      videoNode.currentTime = 0;
+      setIsPlaying(false);
+    };
+    videoNode.addEventListener('ended', onVideoEnd);
+    return () => videoNode.removeEventListener('ended', onVideoEnd);
+  }, [videoNode]);
+
+  return (
+    <Wrapper
+      onPointerEnter={onPointerEnter}
+      onPointerLeave={() => setHovering(false)}
+    >
+      <MediaFrame element={element} />
+      <CSSTransition in={hovering} classNames="button" timeout={200}>
+        <ButtonWrapper key="wrapper" onClick={handlePlayPause}>
```

Comment:
Does this button need a title. Like "Click to play" or something.

Reply:
@spacedmonkey Jonny, the play button is pretty universal . Do you have concerns for accessibility?

Reply:
The title is for screen readers. Blind people will not be able to see the icon of course.
Also consider this screenshot from youtube.  
![Screenshot from 2020-03-25 16-35-08](https://user-images.githubusercontent.com/237508/77561626-dbdcb800-6eb6-11ea-97f5-f4e3ef2e7c5f.png)
On hover a tooltip appears.